File: handler/src/test/java/cc/vileda/kalfor/handler/KalforTestVerticle.java
package cc.vileda.kalfor.handler;

import io.vertx.rxjava.core.AbstractVerticle;
import io.vertx.rxjava.core.http.HttpServer;
import io.vertx.rxjava.ext.web.Router;
import io.vertx.rxjava.ext.web.handler.BodyHandler;
import io.vertx.rxjava.ext.web.handler.CorsHandler;


class KalforTestVerticle extends AbstractVerticle
{
	private final int listenPort;

	KalforTestVerticle(final int listenPort)
	{
		this.listenPort = listenPort;
	}

	@Override
	public void start() throws Exception
	{
		final HttpServer httpServer = vertx.createHttpServer();

		final Router router = Router.router(vertx);
		router.route().handler(CorsHandler.create("*").allowedHeader("authorization"));
		router.route().handler(BodyHandler.create());

		router.route().handler(new SchemaValidationHandler());
		router.route().handler(new CombineHandler(vertx));

		httpServer.requestHandler(router::accept).listen(listenPort);
	}
}

File: library/src/test/kotlin/cc/vileda/kalfor/mock/KalforTestVerticle.kt
package cc.vileda.kalfor.mock

import cc.vileda.kalfor.handler.CombineHandler
import cc.vileda.kalfor.handler.SchemaValidationHandler
import io.vertx.rxjava.core.AbstractVerticle
import io.vertx.rxjava.ext.web.handler.BodyHandler
import io.vertx.rxjava.ext.web.handler.CorsHandler


internal class KalforTestVerticle(private val listenPort: Int) : AbstractVerticle() {
    override fun start() {
        val httpServer = vertx.createHttpServer()

        val router = io.vertx.rxjava.ext.web.Router.router(vertx)
        router.route().handler(CorsHandler.create("*").allowedHeader("authorization"))
        router.route().handler(BodyHandler.create())

        router.post().handler(SchemaValidationHandler())
        router.route().handler(CombineHandler(vertx))

        httpServer.requestHandler({ router.accept(it) }).listen(listenPort)
    }
}
